.chloggen/messaging.consumer.latency.yaml

```
@@ -0,0 +1,22 @@
+# Use this changelog template to create an entry for release notes.
+#
+# If your change doesn't affect end users you should instead start
+# your pull request title with [chore] or use the "Skip Changelog" label.
+
+# One of 'breaking', 'deprecation', 'new_component', 'enhancement', 'bug_fix'
+change_type: enhancement
+
+# The name of the area of concern in the attributes-registry, (e.g. http, cloud, db)
+component: messaging
+
+# A brief description of the change. Surround your text with quotes ("") if it needs to start with a backtick (`).
+note: "Add `messaging.consumer.latency.duration` to capture latency between production and consumption."
```

I suggest to call it ``messaging.consumer.lag`
```suggestion
note: "Add `messaging.consumer.lag` to capture the time difference between when a message was published and when it was consumed."
```
(and correct the brief in the yaml)

---

> I suggest to call it `messaging.consumer.lag`

Let's be careful with that. _Consumer lag_ is commonly defined as a difference of offsets (producer's end offset minus consumer's last committed offset), see example documentation for [Kafka](https://docs.confluent.io/platform/current/monitor/monitor-consumer-lag.html) or [Azure Event Hubs](https://learn.microsoft.com/en-us/samples/azure-samples/eventhub-custom-metrics-emitter/eventhub-custom-metrics-emitter/#overview). Latency is a difference of time stamps.

Both measurements are very important to have, however, to avoid confusion with established terminology what's proposed in this PR shouldn't be called "lag".

---

Agreed, so we need to find a different name, but I still think that `latency.duration` or `latency` would not work and I'd prefer the name that emphasizes that it's a time message spent on the broker before being consumed (this time) rather than in-process latency/duration.

It's common to call it time-in-queue, but we're avoiding queue/topic terminology

---

> but I still think that latency.duration or latency would not work and

I agree. Basically, we're messaging the duration from the end of the "publish" to the beginning of the "process" operation. We don't yet have a satisfying name for this.

The term "enqueued" seems to be a possible candidate (although it messes with our intentions to avoid the term). One could then have `messaging.enqueued.duration` for the latency (time difference), and `messaging.enqueued.count` for the number of unsettled messages in the topic/queue (offset difference).

---

This came up in a few comments.  I tried to address this in my latest changes by declaring two different latencies that represent "creation to processing" time and "enqueued to processing" time. I proposed some specific names: `messaging.latency.duration` and `messaging.buffering.duration` , respective, but I'm open to suggestions.